### PR TITLE
Reviews can be created api

### PR DESCRIPTION
**Pull request by Carrosen**
# [PT Story](https://www.pivotaltracker.com/story/show/166153662)

## Changes we propose in this PR
* Creates unit test for Review model(db_columns, validations, associations, FactoryBot)
* Creates request spec for the reviews controller for post requests
* Creates Review model with following db_columns(score & comment)
* Creates Reviews Controller
* Adds associations between Article and Review model
* Adds published column to article model as boolean
* Adds method for condition (publish_articles) in Reviews controller, updating published param if article reviews is >= 3, and average review score is >= 6

## What we have learned
* Average calculations as conditions in Rails
* Testing request specs for these conditions

### CODE REVIEW

File: app/controllers/api/v1/reviews_controller.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class Api::V1::ReviewsController < ApplicationController
+  after_action :publish_article, only: :create
+
+
```
Comment (oliverochman) [inline app/controllers/api/v1/reviews_controller.rb]: Remove empty line

File: app/controllers/api/v1/reviews_controller.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    def review_params
+      params.require(:review).permit(:score, :comment)
+    end
+
+    def publish_article
+      article = Article.find(params[:article_id])
+      if article.reviews.count >= 3
+        average_score = article.reviews.sum(&:score)/article.reviews.count.to_f
+        if average_score >= 6
+        article.update(published: :true)
```
Comment (oliverochman) [inline app/controllers/api/v1/reviews_controller.rb:29]: indentation

File: spec/requests/api/v1/reviews_spec.rb [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            comment: "You got a lot of typos but apart from that it's a good article :P"
+          }
+        }, headers: headers
+        article.reload
+      end
+
+      it 'does not publish article' do
+        expect(article.published).to eq false
+      end
+
```
Comment (oliverochman) [inline spec/requests/api/v1/reviews_spec.rb]: Remove empty lines

File: app/controllers/api/v1/reviews_controller.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+class Api::V1::ReviewsController < ApplicationController
+  after_action :publish_article, only: :create
+
+
+  def create
+    article = Article.find(params[:article_id])
+    review = article.reviews.create(review_params)
```
Comment (oliverochman) [inline app/controllers/api/v1/reviews_controller.rb]: can someone create a review if the article is already published?

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/reviews_controller.rb (hunk 1 above)
Comment (Carrosen) [reply app/controllers/api/v1/reviews_controller.rb]: done

File: app/controllers/api/v1/reviews_controller.rb (hunk 2 above)
Comment (Carrosen) [reply app/controllers/api/v1/reviews_controller.rb:29]: yes

File: spec/requests/api/v1/reviews_spec.rb (hunk 3 above)
Comment (Carrosen) [reply spec/requests/api/v1/reviews_spec.rb]: yes

**Review by oliverochman — APPROVED** [review]

**Pull request merged**